Add new v4 section and table of contents (#166)
rokucommunity/roku-deploy: README.md

Another set of changes are the names and features available in the Node API. Some have been renamed and others have been change to be used only as CLI commands in order to organize and simplify what is offered. Renamed functions:
- `zipPackage()` -> `zip()`
- `pressHomeButton()` -> `closeChannel()` which will press home twice in order to cancel instant resume
- `publish()` -> `sideload()`
- `signExistingPackage()` -> `createSignedPackage()`
- `deleteInstalledChannel()` -> `deleteDevChannel()`
- `takeScreenshot()` -> `captureScreenshot()`

Some functions were added which allow for any remote-to-Roku interaction: `keyPress()`, `keyUp()`, `keyDown()`, and `sendText()`

Previously, functions `deploy()`, `createPackage()`, and `deployAndSignPackage()` were available in the Node API, but have been moved to CLI commands.

> Lastly, the default files array has changed. node modules and static analysis files have been excluded to speed up load times. Also, `fonts/` and `locale/` was added as they are in a few Roku documentation. The new default array can be seen in the section titled [Files Array](#files-array)

Author: ```suggestion
Lastly, the default files array has changed. node modules and static analysis files have been excluded to speed up load times. Also, `fonts/` and `locale/` was added as they are in some Roku documentation. The new default array can be seen in the section titled [Files Array](#files-array)
```

```

> Can't find what you need? Here are all of the public functions:

Reviewer: Let's add a link to the src/rokudeploy.ts file

- [keyUp()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L187)
- [keyDown()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L194)
- [sendText()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L201)
- [closeChannel()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L226)
- [sideload()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L239)
- [rekeyDevice()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L379)
- [createSignedPackage()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L426)
- [deleteDevChannel()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L614)
- [captureScreenshot()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L629)
- [getOptions()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L698)
- [checkRequiredOptions()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L738)
- [getDeviceInfo()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L787)

> - [getDevId()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L854)

Reviewer: So, I was more thinking you would add a single link to `RokuDeploy.ts`, becuase these line numbers will change as we update the code. I think you can just list the functions without links, with a single link at the top saying "look through this file to see all the public methods available"

Author: Ohhh ok. Yeah the line numbers would change so one link sounds easier to maintain

Author: Done and also changed think to typescript interface for Roku Deploy Options
